core/spec/models/spree/preferences/preferable_spec.rb

```diff
@@ -151,158 +156,170 @@ def self.allowed_admin_form_preference_types
-        expect(@a.preferences[:product_attributes]).to eq({ id: 1, name: 2 })
+        config_class_a.preference :product_ids, :any, default: []
+        config_class_a.preference :product_attributes, :any, default: {}
+
+        expect(a.preferences[:product_attributes]).to eq({})
+        a.set_preference(:product_attributes, { id: 1, name: 2 })
+        expect(a.preferences[:product_attributes]).to eq({ id: 1, name: 2 })
       end
     end
 
     context "converts encrypted_string preferences to encrypted values" do
       it "with string, encryption key provided as option" do
-        A.preference :secret, :encrypted_string,
+        config_class_a.preference :secret, :encrypted_string,
                      encryption_key: 'VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!'
 
-        @a.set_preference(:secret, 'secret_client_id')
-        expect(@a.get_preference(:secret)).to eq('secret_client_id')
-        expect(@a.preferences[:secret]).not_to eq('secret_client_id')
+        a.set_preference(:secret, 'secret_client_id')
+        expect(a.get_preference(:secret)).to eq('secret_client_id')
+        expect(a.preferences[:secret]).not_to eq('secret_client_id')
       end
 
       it "with string, encryption key provided as env variable" do
         expect(ENV).to receive(:[]).with("SOLIDUS_PREFERENCES_MASTER_KEY").and_return("VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!")
 
-        A.preference :secret, :encrypted_string
+        config_class_a.preference :secret, :encrypted_string
 
-        @a.set_preference(:secret, 'secret_client_id')
-        expect(@a.get_preference(:secret)).to eq('secret_client_id')
-        expect(@a.preferences[:secret]).not_to eq('secret_client_id')
+        a.set_preference(:secret, 'secret_client_id')
+        expect(a.get_preference(:secret)).to eq('secret_client_id')
+        expect(a.preferences[:secret]).not_to eq('secret_client_id')
       end
 
       it "with string, encryption key provided as option, set using syntactic sugar method" do
-        A.preference :secret, :encrypted_string,
+        config_class_a.preference :secret, :encrypted_string,
                      encryption_key: 'VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!'
 
-        @a.preferred_secret = 'secret_client_id'
-        expect(@a.preferred_secret).to eq('secret_client_id')
-        expect(@a.preferences[:secret]).not_to eq('secret_client_id')
+        a.preferred_secret = 'secret_client_id'
+        expect(a.preferred_secret).to eq('secret_client_id')
+        expect(a.preferences[:secret]).not_to eq('secret_client_id')
       end
 
       it "with string, default value" do
-        A.preference :secret, :encrypted_string,
+        config_class_a.preference :secret, :encrypted_string,
                      default: 'my_default_secret',
                      encryption_key: 'VkYp3s6v9y$B?E(H+MbQeThWmZq4t7w!'
 
-        expect(@a.get_preference(:secret)).to eq('my_default_secret')
-        expect(@a.preferences[:secret]).not_to eq('my_default_secret')
+        a = A.new
```

Comment:
```suggestion
        a = config_class_a.new
```

I think that'll make tests green